src/helper/UserSession.cpp

```diff
@@ -174,7 +174,6 @@ namespace SDDM {
                 if (ioctl(STDIN_FILENO, TIOCSCTTY) < 0) {
                     const auto error = strerror(errno);
                     qCritical().nospace() << "Failed to take control of " << ttyString << " (" << QFileInfo(ttyString).owner() << "): " << error;
-                    exit(Auth::HELPER_OTHER_ERROR);
```

Comment:
>Things still work,

Based on what?

If we don't do this how can rootless xorg still call vt_activate without errors?

/If/ that isn't needed anymore why do we have this line in the first place?

--

Interestingly gdm-session-worker does not abort if TIOCSCTTY fails. It also

Reply:
> Based on what?

Based on reproducing the issue with the patch applied and having the session work again.

> If we don't do this how can rootless xorg still call vt_activate without errors?
> 
> /If/ that isn't needed anymore why do we have this line in the first place?

We call rootless x with -novtswitch and sddm does the switching, so yes, rootless x doesn't need it. I see Wayland compositors do trigger a switch, but we also switch for them so it shouldn't be a problem either.

We might be able to remove that code, maybe @plfiorini can tell us why it was added first.

Reply:
Removed this change in order to move forward.